Prevent skeleton display when express payment methods are active

ralucaStan opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

- We're only displaying Express Payments skeleton when an express payment isn't active to prevent the buttons to rerender. I've also added the fix to the cart block
- I've improved the a11y handling in both cases
- There is some duplication in place in the files, but those were there before [the PR](https://github.com/woocommerce/woocommerce/pull/59449) that introduced the bug. This duplication can be addressed outside of this PR

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

Closes https://github.com/woocommerce/woocommerce/issues/60362
Closes https://linear.app/a8c/issue/WOOPLUG-5315/checkout-block-express-payment-components-unmount-when-shipping-rates

(For Bug Fixes) Bug introduced in PR  https://github.com/woocommerce/woocommerce/pull/59449


### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

If you want to use PayPal Payments in your local store
- Clone locally https://github.com/woocommerce/woocommerce-paypal-payments
- Go to WooCommerce > Payments > PayPal
- Go through setup wizzard
- Choose test sandbox
- Create a developer sandbox account [Instructions for the account here](https://github.com/woocommerce/woocommerce-paypal-payments/wiki/Testing-Setup) (don’t try to log in with personal account, that you use as a shopper)


## Checkout block
1. Go to Checkout block
2. Checkout using the Express PayPal
3. In the PayPal window do some address manipulation
4. Place the order without issues

## Cart block - will be handled separately
Progressive loading for Cart comes in 10.2; I'll update the logic in a separate before the feature freeze of 10.2


Optionally also repeat tests with Stripe, but the bug was not happening there.

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

I've updated the checkout testing with this case;
Added tests for the cart express area, there weren't any in place

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

ralucaStan committed:
Prevent skeleton display when express payment methods are active

ralucaStan committed:
Add changelog

ralucaStan commented on plugins/woocommerce/client/blocks/assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/checkout-express-payment.tsx:
```diff
@@ -93,18 +102,18 @@ const CheckoutExpressPayment = () => {
-					'Processing express checkout',
-					'woocommerce'
-				) }
+				{ ...( checkoutProcessing && {
```
These were always displayed, I've improved the logic and added testing for it.

ralucaStan committed:
Remove checkout process related disabled states

ralucaStan committed:
Rename variable for better clarity

ralucaStan committed:
Merge branch 'trunk' into wooplug-5315-checkout-block-express-payment-components-unmount-when

straku commented on plugins/woocommerce/client/blocks/assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/cart-express-payment.js:
```diff
@@ -50,6 +50,14 @@ const CartExpressPayment = () => {
 		registeredExpressPaymentMethods,
 	} );
 
+	// We show the skeleton when
+	// the express payment method is not active (because they trigger recalculations) and
+	// the checkout is calculating, because it can result in different express payment methods
+	// or when the express payment methods are not initialized
```
```suggestion
	// We show the skeleton when the express payment method is not active
	// (because they trigger recalculations) and the checkout is calculating,
	// because it can result in different express payment methods or when the
	// express payment methods are not initialized
```
Found the first line weirdly short, reflowed with claude

ralucaStan committed:
Update plugins/woocommerce/changelog/wooplug-5315-checkout-block-express-payment-components-unmount-when

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

ralucaStan commented on plugins/woocommerce/client/blocks/assets/js/blocks/cart-checkout-shared/payment-methods/express-payment/cart-express-payment.js:
```diff
@@ -50,6 +50,14 @@ const CartExpressPayment = () => {
+	// or when the express payment methods are not initialized
+	const showSkeleton =
+		! isExpressPaymentMethodActive &&
+		( isCalculating || hasRegisteredNotInitializedExpressPaymentMethods );
```
I'll actually remove isCalculating which is checkout specific and only include `hasPendingOperations`.
The only available actions in cart: quantity update & coupons. In checkout, we don't reload for coupons, so this simplifies things.

ralucaStan committed:
Update cart logic to only account for cart item actions and not for checkout related flags

ralucaStan committed:
Undo cart changes, these will be added in a separate PR
We're only fixing the checkout behaviour so that we can target the 10.1 release

opr reviewed (approved):
I tested this and it's working for me - the code changes look fine.

I also tested with Google Pay and Link from Stripe and verified that changing address there did not break the experience.
